js/ui/control/attribution.js

```diff
@@ -49,6 +49,15 @@ Attribution.prototype = util.inherit(Control, {
             }
         }
 
+        // remove any entries that are duplicates or strict substrings of another entry.
+        attributions.sort();
```

Comment:
> you're comparing every attribution string to every other attribution string

Agreed, but it's also directional, since we're looking for a substring match and want to keep the longer of the two if there's a match.

I agree the sort isn't necessary, but removing it does make the logic in the filter function a little more particular... I think it would then need to be something like:

```js
attribution = attribution.filter(function (a, i) {
  // check if `a` is a substring of another string in the list
  var isSubsumedByAnother = attribution.some(function (b, j) {
    // skip if we're checking the same element
    if (i === j) { return false }
    // if a === b, then only call it a match if a's index is higher; otherwise
    // we would filter out both elements when there's an exact match
    if (a === b) { return i > j }
    // check if a is a substring of b
    if (b.indexOf(a) >= 0) { return true }
  })
  return !isSubsumedByAnother
})
```

@lucaswoj I considered this, and ended up feeling that both approaches were equally bad... lemme know if you find this way clearer and I can change it